include/zephyr/bluetooth/settings.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief This function act as a garbage collector for the Bluetooth settings.
+ * It will:
+ * - Remove not used settings key under 'bt/' namespace;
+ * - Remove possible leftover from unexpected behavior.
+ *
+ * The settings are removed directly from the storage, the user need to run @ref
+ * settings_load (or @ref settings_load_subtree to only load 'bt' namespace) to
+ * apply the changes.
+ *
+ * Example of possible leftover: An identity has been deleted, the corresponding
+ * address as been removed from 'bt/id' but before removing the corresponding
+ * key in 'bt/keys', the device has been powered off. Calling
+ * `bt_settings_cleanup` will ensure that the key is removed.
```

Comment:
Would it make sense to have an additional function that verifies the settings to indicate whether calling this function will change anything? 

Something along the lines of `bt_settings_valid` so that the use of this function will be
```c
if (!bt_settings_valid()) {
    bt_settings_cleanup();
}
```

or some other way of doing a dry run? I could imagine that for some it may be interesting to see what is incorrect / will be cleaned up before this is called.

Reply:
> I am not sure of the use case. 

If I am a user and I have my settings stored in flash, I'd perhaps rather be aware of an issue in my settings before deleting them, rather than just silently deleting them, e.g. if I store my own settings and that I am foolish enough to store something under `bt/id` myself, then I'd rather become aware of that, rather than it just silently getting deleted by the stack